File: algorithms/398. Random Pick Index/solution.py
class Solution(object):

    def __init__(self, nums):
        """
        :type nums: List[int]
        :type numsSize: int
        """
        self.nums = nums

    def pick(self, target):
        """
        :type target: int
        :rtype: int
        """
        cnt = 0
        index = -1
        for i, n in enumerate(self.nums):
            if target == n:
                if random.randint(0, cnt) == 0:
                    index = i
                cnt += 1
        return index


# Your Solution object will be instantiated and called as such:
# obj = Solution(nums)
# param_1 = obj.pick(target)

File: algorithms/398. Random Pick Index/solution.cc
class Solution {
    vector<int> nums;
public:
    Solution(vector<int> nums) {
        this->nums = nums;
    }
    
    int pick(int target) {
        int idx;
        int count = 1;
        for ( int i = 0; i < nums.size(); i++ ) {
            if ( target == nums.at(i) ) {
                if ( rand() % count == 0 ) {
                    idx = i;
                }
                count += 1;
            }
        }
        return idx;
    }
};

/**
 * Your Solution object will be instantiated and called as such:
 * Solution obj = new Solution(nums);
 * int param_1 = obj.pick(target);
 */
